File: presentation/src/main/java/com/hnh/presentation/util/ConstantExercise.kt
package com.hnh.presentation.util

object LOGIC{
    const val LOGIC = "Logic"
    const val READY = "Ready"
    const val ACTION = "Action"
}
object EXERCISE_INFORMATION{
    //세트, 운동상태 휴식시간 상수
    const val SET_FIRST_INT = 1
    const val SET_SECOND_INT  = 2
    const val SET_THIRD_INT = 3
    const val SET_FORTH_INT = 4
    const val SET_FIFTH_INT = 5
    const val DEFAULT_REST_TIME = 40000
    const val STATE_EXERCISE_STOP = 0
    const val STATE_EXERCISE_START = 1
    const val STATE_REST = 2
    const val STATE_EXERCISE_FINISH = 3
    // 운동정보 전체 사용 상수
    const val UPPER = 0
    const val LOWER = 1
    const val FULL = 2
    const val LONG_TIME = 1000
    const val MID_TIME = 800
    const val SHOUT_TIME = 600
    const val REPS_DEFAULT = 20
    const val SET_DEFAULT = 3
    const val REPS_LONG = 30
    const val SET_LONG = 5
    const val REPS_SHOUT = 15
    const val SET_SHOUT = 2
    const val FIRST_SET = 99
    const val MODE_SET = 0
    const val MODE_SELECT_SET = 1
    const val MODE_FREE = 2

    const val SETTING_MODE = 0
    const val SETTING_SET = 1
    const val SETTING_REPS = 2
    const val SETTING_REST_TIME = 3

}
object EXERCISE_LIST{
    const val DEAD_LIFT = "DeadLift"
    const val SQUAT = "Squat"
    const val SIDE_LATERAL_RAISE = "SideLateralRaise"
    const val SHOULDER_PRESS = "ShoulderPress"
    const val LAT_PULL_DOWN = "LatPullDownBack"
    const val CABLE_ROW = "CableRow"
    const val LUNGE = "Lunge"
    const val CABLE_PUSH_DOWN = "CablePushDown"
    const val KICK_BACK_DUMBBELL = "KickBackDumbbell"
    const val STANDING_BARBELL_CURL = "StandingBarbellCurl"
    const val LEG_RAISE = "LegRaise"

    fun constrainList(exerciseID: String) : Boolean{
        val exerciseList = arrayListOf<String>()
        var result = false
        exerciseList.add(DEAD_LIFT)
        exerciseList.add(SQUAT)
        exerciseList.add(SIDE_LATERAL_RAISE)
        exerciseList.add(SHOULDER_PRESS)
        exerciseList.add(LAT_PULL_DOWN)
        exerciseList.add(CABLE_ROW)
        exerciseList.add(LUNGE)
        exerciseList.add(CABLE_PUSH_DOWN)
        exerciseList.add(KICK_BACK_DUMBBELL)
        exerciseList.add(STANDING_BARBELL_CURL)
        for(ID in exerciseList){
            if(ID == exerciseID){
                result =  true
            }
        }
        return result
    }
}

object DEAD_LIFT{
    const val FOOT_DISTANCE = "FootDistance"
    const val HIP_ANGLE_DOWN = "HipAngleDown"
    const val HIP_ANGLE_UP = "HipAngleUp"
    const val BAR_WRIST = "BarWrist"
}

object SQUAT{
    const val FOOT_TO_SHOULDER_DISTANCE = "FootToShoulderDistance"
    const val KNEE_POINT = "KneePoint"
    const val KNEE_DISTANCE = "KneeDistance"
    const val HIP_ANGLE = "HipAngle"
    const val KNEE_ANGLE_DOWN = "KneeAngleDown"
}

object SIDE_LATERAL_RAISE{
    const val WRIST_3D = "Wrist3D"
    const val SHOULDER_TO_WRIST = "ShoulderToWrist"
    const val ELBOW_ANGLE = "ElbowAngle"
    const val HIP_ANGLE = "HipAngle"
    const val ELBOW_ANGLE_DOWN = "ElbowAngleDown"
    const val ELBOW_ANGLE_UP = "ElbowAngleUp"
}

object SHOULDER_PRESS{
    const val DOWN_WRIST = "DownWrist"
    const val WRIST_3D = "Wrist3D"
    const val ELBOW_ANGLE = "ElbowAngle"
    const val ELBOW_UP = "ElbowUp"
    const val ELBOW_DOWN = "ElbowDown"
    const val ELBOW_FRONT = "ElbowFront"
}

object LAT_PULL_DOWN{
    const val WRIST_BAR = "BarWrist"
    const val ELBOW_VERTICAL = "ElbowVertical"
    const val WRIST_DOWN = "WristDown"
    const val ELBOW_UP = "ElbowUp"
    const val HIP_ANGLE = "HipAngle"
}

object CABLE_ROW{
}

object LUNGE{
    const val KNEE_ANGLE_DOWN = "KneeAngleDown"
    const val FOOT_TO_PELVIS_DISTANCE = "FootToPelvisDistance"
    const val KNEE_POINT = "LungeKneePoint"
    const val KNEE_TO_CENTER = "KneeToCenter"
    const val HIP_ANGLE = "HipAngle"
    const val FOOT_STRAIGHT = "FootStraight"
}

object CABLE_PUSH_DOWN{
    const val ELBOW_FIX = "ElbowFix"
    const val HIP_FIX = "HipFix"
    const val ELBOW_ANGLE_DOWN = "ElbowAngleDown"
    const val ELBOW_ANGLE_UP = "ElbowAngleUp"
}

object KICK_BACK_DUMBBELL{
}

object STANDING_BARBELL_CURL{
    const val ELBOW_FIX = "ElbowFix"
    const val HIP_FIX = "HipFix"
    const val ELBOW_ANGLE_DOWN = "ElbowAngleDown"
    const val ELBOW_ANGLE_UP = "ElbowAngleUp"
}

object LEG_RAISE{
    const val HIP_ANGLE_DOWN = "HipAngleDown"
    const val HIP_ANGLE_UP = "HipAngleUp"
}

File: iOS/FitGAI/PoseLogic/ConstantExercise.swift
//
//  ConstantExercise.swift
//  FitGAI
//
//  Created by 김기훈 on 2021/12/13.
//

import Foundation

class LOGIC {
    static let LOGIC = "Logic"
    static let READY = "Ready"
    static let ACTION = "Action"
}

class EXERCISE_INFORMATION{
    //세트, 운동상태 휴식시간 상수
    
    static var DEFAULT_REST_TIME:Int64 = Int64(40000)
    static var REPS_DEFAULT:Int = 20

    static let SET_FIRST_INT:Int32 = 1
    static let SET_SECOND_INT:Int32  = 2
    static let SET_THIRD_INT:Int32 = 3
    static let SET_FORTH_INT:Int32 = 4
    static let SET_FIFTH_INT:Int32 = 5
    static let STATE_EXERCISE_STOP:Int32 = 0
    static let STATE_EXERCISE_START:Int32 = 1
    static let STATE_REST:Int32 = 2
    // 운동정보 전체 사용 상수
    static let UPPER = 0
    static let LOWER = 1
    static let FULL = 2
    static let LONG_TIME = Int64(1000)
    static let MID_TIME = Int64(800)
    static let SHOUT_TIME = Int64(600)
    static let SET_DEFAULT = 3
    static let REPS_LONG = 30
    static let SET_LONG = 5
    static let REPS_SHOUT = 15
    static let SET_SHOUT = 2
    static let FIRST_SET = 99
    static let MODE_SET = 0
    static let MODE_SELECT_SET = 1
    static let MODE_FREE = 2
    
    class func initializeVariables() {
        switch FeatureConst.FEATURE_APP_LEVEL {
        case .debug:
            DEFAULT_REST_TIME = Int64(5000)
            REPS_DEFAULT = 3
        case .release:
            DEFAULT_REST_TIME = Int64(40000)
            REPS_DEFAULT = 20
        case .demo:
            DEFAULT_REST_TIME = Int64(15000)
            REPS_DEFAULT = 5
        }
    }
}

class DEAD_LIFT {
    static let DEAD_LIFT = "DeadLift"
    static let FOOT_DISTANCE = "FootDistance"
    static let HIP_ANGLE_DOWN = "HipAngleDown"
    static let HIP_ANGLE_UP = "HipAngleUp"
    static let BAR_WRIST = "BarWrist"
}

class SQUAT {
    static let SQUAT = "Squat"
    static let FOOT_TO_SHOULDER_DISTANCE = "FootToShoulderDistance"
    static let KNEE_POINT = "KneePoint"
    static let KNEE_DISTANCE = "KneeDistance"
    static let HIP_ANGLE = "HipAngle"
    static let KNEE_ANGLE_DOWN = "KneeAngleDown"
}

class SIDE_LATERAL_RAISE {
    static let SIDE_LATERAL_RAISE = "SideLateralRaise"
    static let WRIST_3D = "Wrist3D"
    static let SHOULDER_TO_WRIST = "ShoulderToWrist"
    static let ELBOW_ANGLE = "ElbowAngle"
    static let HIP_ANGLE = "HipAngle"
    static let ELBOW_ANGLE_DOWN = "ElbowAngleDown"
    static let ELBOW_ANGLE_UP = "ElbowAngleUp"
}

class SHOULDER_PRESS {
    static let SHOULDER_PRESS = "ShoulderPress"
    static let DOWN_WRIST = "DownWrist"
    static let WRIST_3D = "Wrist3D"
    static let ELBOW_ANGLE = "ElbowAngle"
    static let ELBOW_UP = "ElbowUp"
    static let ELBOW_DOWN = "ElbowDown"
    static let ELBOW_FRONT = "ElbowFront"
}

class LAT_PULL_DOWN {
    static let LAT_PULL_DOWN = "LatPullDownBack"
    static let WRIST_BAR = "BarWrist"
    static let ELBOW_VERTICAL = "ElbowVertical"
    static let WRIST_DOWN = "WristDown"
    static let ELBOW_UP = "ElbowUp"
    static let HIP_ANGLE = "HipAngle"
}

class CABLE_LOW {
    static let CABLE_LOW = "CableLow"
}

class LUNGE {
    static let LUNGE = "Lunge"
    static let KNEE_ANGLE_DOWN = "KneeAngleDown"
    static let FOOT_TO_PELVIS_DISTANCE = "FootToPelvisDistance"
    static let KNEE_POINT = "LungeKneePoint"
    static let KNEE_TO_CENTER = "KneeToCenter"
    static let HIP_ANGLE = "HipAngle"
    static let FOOT_STRAIGHT = "FootStraight"
}

class CABLE_PUSH_DOWN {
    static let CABLE_PUSH_DOWN = "CablePushDown"
    static let ELBOW_FIX = "ElbowFix"
    static let HIP_FIX = "HipFix"
    static let ELBOW_ANGLE_DOWN = "ElbowAngleDown"
    static let ELBOW_ANGLE_UP = "ElbowAngleUp"
}

class KICK_BACK_DUMBBELL {
    static let KICK_BACK_DUMBBELL = "KickBackDumbbell"
}

class STANDING_BARBELL_CURL {
    static let STANDING_BARBELL_CURL = "StandingBarbellCurl"
    static let ELBOW_FIX = "ElbowFix"
    static let HIP_FIX = "HipFix"
    static let ELBOW_ANGLE_DOWN = "ElbowAngleDown"
    static let ELBOW_ANGLE_UP = "ElbowAngleUp"
}
